src/server/snapshot.cc

```diff
@@ -377,6 +435,9 @@ void SliceSnapshot::OnDbChange(DbIndex db_index, const DbSlice::ChangeReq& req)
       stats_.side_saved += SerializeBucket(db_index, it);
     });
   }
+
+  // Flush tiered delayed entries to avoid reordering with journal
+  tiered_serializer_->FlushDelayed(true);
 }
```

Comment:
Ideally we pass the value with ChangeReq& and find out whether it was tiered, only if it's tiered and part of a relevant page, we should flush it